subsys/bluetooth/audio/pacs.c

```diff
@@ -499,96 +519,157 @@ static ssize_t src_loc_write(struct bt_conn *conn,
 		return BT_GATT_ERR(BT_ATT_ERR_WRITE_REQ_REJECTED);
 	}
 
-	err = set_src_location(location);
-	if (err != 0) {
-		LOG_DBG("write_location returned %d", err);
-		return BT_GATT_ERR(BT_ATT_ERR_WRITE_REQ_REJECTED);
-	}
+	set_src_location(location);
 
 	return len;
 }
 #endif /* CONFIG_BT_PAC_SRC_LOC_WRITEABLE */
 
+
+static sys_slist_t *pacs_get(enum bt_audio_dir dir)
+{
+	switch (dir) {
+#if defined(CONFIG_BT_PAC_SNK)
+	case BT_AUDIO_DIR_SINK:
+		return &snk_pacs_list;
+#endif /* CONFIG_BT_PAC_SNK */
+#if defined(CONFIG_BT_PAC_SRC)
+	case BT_AUDIO_DIR_SOURCE:
+		return &src_pacs_list;
+#endif /* CONFIG_BT_PAC_SRC */
+	default:
+		return NULL;
+	}
+}
+
 BT_GATT_SERVICE_DEFINE(pacs_svc,
 	BT_GATT_PRIMARY_SERVICE(BT_UUID_PACS),
 #if defined(CONFIG_BT_PAC_SNK)
+#if defined(CONFIG_BT_PAC_SNK_NOTIFIABLE)
 	BT_AUDIO_CHRC(BT_UUID_PACS_SNK,
 		      BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY,
 		      BT_GATT_PERM_READ_ENCRYPT,
 		      snk_read, NULL, NULL),
 	BT_AUDIO_CCC(snk_cfg_changed),
+#else
+	BT_AUDIO_CHRC(BT_UUID_PACS_SNK,
+		      BT_GATT_CHRC_READ,
+		      BT_GATT_PERM_READ_ENCRYPT,
+		      snk_read, NULL, NULL),
+#endif /* CONFIG_BT_PAC_SNK_NOTIFIABLE */
 #if defined(CONFIG_BT_PAC_SNK_LOC)
-#if defined(CONFIG_BT_PAC_SNK_LOC_WRITEABLE)
+#if defined(CONFIG_BT_PAC_SNK_LOC_WRITEABLE) && defined(CONFIG_BT_PAC_SNK_LOC_NOTIFIABLE)
 	BT_AUDIO_CHRC(BT_UUID_PACS_SNK_LOC,
 		      BT_GATT_CHRC_READ | BT_GATT_CHRC_WRITE | BT_GATT_CHRC_NOTIFY,
 		      BT_GATT_PERM_READ_ENCRYPT | BT_GATT_PERM_WRITE_ENCRYPT,
 		      snk_loc_read, snk_loc_write, NULL),
-#else
+	BT_AUDIO_CCC(snk_loc_cfg_changed),
+#elif defined(CONFIG_BT_PAC_SNK_LOC_NOTIFIABLE)
 	BT_AUDIO_CHRC(BT_UUID_PACS_SNK_LOC,
 		      BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY,
 		      BT_GATT_PERM_READ_ENCRYPT,
 		      snk_loc_read, NULL, NULL),
-#endif /* CONFIG_BT_PAC_SNK_LOC_WRITEABLE */
 	BT_AUDIO_CCC(snk_loc_cfg_changed),
+#elif defined(CONFIG_BT_PAC_SNK_LOC_WRITEABLE)
+	BT_AUDIO_CHRC(BT_UUID_PACS_SNK_LOC,
+		      BT_GATT_CHRC_READ | BT_GATT_CHRC_WRITE,
+		      BT_GATT_PERM_READ_ENCRYPT,
+		      snk_loc_read, snk_loc_write, NULL),
+#else
+	BT_AUDIO_CHRC(BT_UUID_PACS_SNK_LOC,
+		      BT_GATT_CHRC_READ,
+		      BT_GATT_PERM_READ_ENCRYPT,
+		      snk_loc_read, NULL, NULL),
+#endif /* (CONFIG_BT_PAC_SNK_LOC_WRITEABLE && CONFIG_BT_PAC_SNK_LOC_NOTIFIABLE) */
```

Comment:
It's a bit hard to read (and maintain) IMO. Maybe consider using some macros e.g.
```
#define BT_PACS_SNK_LOC_PERM \
        BT_GATT_CHRC_READ \
        IF_ENABLED(CONFIG_BT_PAC_SNK_LOC_WRITEABLE, (| BT_GATT_CHRC_WRITE), ())
#define BT_PACS_SNK_LOC_PROP \
        BT_GATT_CHRC_READ \
        IF_ENABLED(CONFIG_BT_PAC_SNK_LOC_WRITEABLE, (| BT_GATT_CHRC_WRITE), ()) \
        IF_ENABLED(CONFIG_BT_PAC_SNK_LOC_NOTIFIABLE, (| BT_GATT_CHRC_NOTIFY), ())
#define BT_PACS_SNK_LOC(_read, _write) \
	BT_AUDIO_CHRC(BT_UUID_PACS_SNK_LOC, BT_PACS_SNK_LOC_PROP, \
                      BT_PACS_SNK_LOC_PERM, _read, _write, NULL), \
        IF_ENABLED(CONFIG_BT_PAC_SNK_LOC_NOTIFIABLE, (BT_AUDIO_CCC(snk_loc_cfg_changed),), ())
```

applies to all characteristics

Reply:
Let's leave it as it is. We can refactor this later.

Reply:
I admit I was wrong, what's your opinion?

Reply:
I have already made this change :D Please take a look.
I think it turns out to be more readable indeed.